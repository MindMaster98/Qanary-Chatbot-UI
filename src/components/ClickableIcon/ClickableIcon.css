.clickableIcon {
    border-radius: 75px;
    overflow: hidden;
    cursor: pointer;
    flex-shrink: 0;
}

.clickableIcon:hover, .clickableIcon:focus {
    background-color: rgba(255, 255, 255, 0.5);
    outline: none;
}

.clickableIcon img {
    position: relative;
    top: 20%;
    left: 20%;
    height: 60%;
}
