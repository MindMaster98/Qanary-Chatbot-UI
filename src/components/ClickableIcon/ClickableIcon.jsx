import React from "react";
import texts from "../../texts/de/texts.json";
import "./ClickableIcon.css";

/**
 * 
 * @property {function} onClick The function that gets executed when the button is clicked
 * @property {string} icon The URI of the image that should be used as the icon
 * @property {string} alt The alt text that should be displayed, if the image cannot be loaded
 * @property {object} style The style of the div containing the img element
 */
export default function ClickableIcon(props) {
  return (
    <div
      className={"clickableIcon " + (props.className ?? "")}
      onClick={props.onClick ?? undefined}
      style={props.style ?? {}}
      tabIndex={props.tabIndex ?? "0"}
    >
      <img
        tabIndex="-1"
        src={props.icon ?? `https://plchldr.co/i/50x50?&bg=8f8e8e&fc=000&text=${texts["clickable-icon"]["fallback-text"]}`}
        alt={props.alt ?? texts["clickable-icon"]["fallback-text"]}
      />
    </div>
  );
}
