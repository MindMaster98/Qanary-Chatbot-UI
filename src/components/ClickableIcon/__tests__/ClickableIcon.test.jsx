import renderer from "react-test-renderer";
import { act } from "@testing-library/react";
import ReactDOM from "react-dom";
import ClickableIcon from "../ClickableIcon";

// Snapshot tests

it("renders correctly with no props", () => {
  const tree = renderer.create(<ClickableIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Blackbox tests
// You cannot dispatch events using test renderer, so I need to actually add the element to the DOM 
// and dispatch an actual event. Calling the function directly would oppose the idea of a blackbox.

let container;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

it("calls the onClick function when clicked", () => {
  const mockFn = jest.fn();

  act(() => {
    ReactDOM.render(<ClickableIcon onClick={mockFn} />, container);
  });

  const div = document.querySelector(".clickableIcon");
  div.dispatchEvent(new MouseEvent("click", { bubbles: true }));

  expect(mockFn).toHaveBeenCalledTimes(1);
});

it("does not call the onClick function when not clicked", () => {
  const mockFn = jest.fn();

  act(() => {
    ReactDOM.render(<ClickableIcon onClick={mockFn} />, container);
  });

  expect(mockFn).toHaveBeenCalledTimes(0);
});