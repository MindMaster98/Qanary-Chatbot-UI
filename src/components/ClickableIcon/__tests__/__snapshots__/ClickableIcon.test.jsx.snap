// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no props 1`] = `
<div
  className="clickableIcon "
  style={Object {}}
  tabIndex="0"
>
  <img
    alt="Kein Icon gefunden"
    src="https://plchldr.co/i/50x50?&bg=8f8e8e&fc=000&text=Kein Icon gefunden"
    tabIndex="-1"
  />
</div>
`;
